crates/router/src/types/storage/payment_intent.rs

```diff
@@ -40,7 +40,7 @@ impl PaymentIntentDbExt for PaymentIntent {
         // when https://github.com/rust-lang/rust/issues/52662 becomes stable
         let mut filter = <Self as HasTable>::table()
             .filter(dsl::merchant_id.eq(merchant_id.to_owned()))
-            .order(dsl::modified_at.desc())
+            .order(dsl::created_at.desc())
```

Comment:
The intent were sorting based on modified_at before, I'm not sure if changing this will cause issues elsewhere (I have tested in postman in local and payments list is working as intended)

Reply:
👍 Yeah this should be fine...